x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/sagemaker/schema/openai/OpenAiTextEmbeddingPayload.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.sagemaker.schema.openai;
+
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.sagemakerruntime.model.InvokeEndpointResponse;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.inference.InferenceServiceResults;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xcontent.XContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.json.JsonXContent;
+import org.elasticsearch.xpack.inference.services.openai.response.OpenAiEmbeddingsResponseEntity;
+import org.elasticsearch.xpack.inference.services.sagemaker.SageMakerInferenceRequest;
+import org.elasticsearch.xpack.inference.services.sagemaker.model.SageMakerModel;
+import org.elasticsearch.xpack.inference.services.sagemaker.schema.SageMakerSchemaPayload;
+import org.elasticsearch.xpack.inference.services.sagemaker.schema.SageMakerStoredServiceSchema;
+import org.elasticsearch.xpack.inference.services.sagemaker.schema.SageMakerStoredTaskSchema;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalPositiveInteger;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalString;
+
+public class OpenAiTextEmbeddingPayload implements SageMakerSchemaPayload {
+
+    private static final XContent jsonXContent = JsonXContent.jsonXContent;
+    private static final String APPLICATION_JSON = jsonXContent.type().mediaTypeWithoutParameters();
+
+    @Override
+    public String api() {
+        return "openai";
+    }
+
+    @Override
+    public EnumSet<TaskType> supportedTasks() {
+        return EnumSet.of(TaskType.TEXT_EMBEDDING);
+    }
+
+    @Override
+    public SageMakerStoredServiceSchema extraServiceSettings(Map<String, Object> serviceSettings, ValidationException validationException) {
```

Comment:
Oh I see `extraServiceSettings` is the `api` specific service settings 👍 

nit: I think it might have been more clear if it something like that `apiServiceSettings`.